app/models/sign_in/concerns/certifiable.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module SignIn
+  module Concerns
+    module Certifiable
+      extend ActiveSupport::Concern
+
+      def assertion_public_keys
+        @assertion_public_keys ||= certificate_objects.map(&:public_key)
+      end
+
+      def certificate_with_serial(serial)
+        certificate_objects.find do |certificate|
+          certificate.serial.to_s == serial
+        end
+      end
+
+      def delete_certificate_with_serial(serial)
```

Comment:
I don't see any usage of this function, can probably 🪓 it

Reply:
I implemented this so we had a convenient way to delete invalid certificates. It's outside of the PR's scope, but I thought it made sense to include it. After all, we'll get notifications for these invalid certificates and will probably need to do something to clean them up.